src/equals.ts

```diff
@@ -0,0 +1,41 @@
+/** @module ssz */
+import assert from "assert";
+import BN from "bn.js";
+
+import {
+  AnySSZType,
+  FullSSZType,
+  Type,
+} from "./types";
+import { assertValidValue } from "./assertValidValue";
+import { parseType } from "./util/types";
+
+
+/**
+ * Equality of two values of an SSZ type
+ */
+export function equals(value1: any, value2: any, type: AnySSZType): boolean {
+  const _type = parseType(type);
+  assertValidValue(value1, _type);
+  assertValidValue(value2, _type);
+  return _equals(value1, value2, _type);
+}
+
+function _equals(value1: any, value2: any, type: FullSSZType): boolean {
+  switch (type.type) {
+    case Type.uint:
+      return (new BN(value1)).eq(new BN(value2));
+    case Type.bool:
+      return value1 === value2;
+    case Type.byteList:
```

Comment:
intentional skip because `byteList` is effectively the same as `byteVector`?

Reply:
Yea for both we want to do the same thing.

Reply:
is that a property of `switch`, if a case is defined with no actual statement, then it automatically triggers the next?

Reply:
Yeah, thats one of the tricky things about `switch`. Check out: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#What_happens_if_I_forgot_a_break

Reply:
weird